codegen/type_converter.go

```diff
@@ -638,25 +689,61 @@ func (c *TypeConverter) genStructConverter(
 			var (
 				stFromPrefix = keyPrefix
 				stFromType   compile.TypeSpec
+				fromTypeName string
 			)
 			if fromField != nil {
 				stFromType = fromField.Type
 				stFromPrefix = keyPrefix + pascalCase(fromField.Name)
+
+				var err error
+				fromTypeName, err = c.getIdentifierName(stFromType)
+				if err != nil {
+					fromTypeName = ""
+				}
 			}
-			err := c.genConverterForStruct(
-				toField.Name,
-				toFieldType,
-				toField.Required,
-				stFromType,
-				fromIdentifier,
-				keyPrefix+pascalCase(toField.Name),
-				stFromPrefix,
-				indent,
-				fieldMap,
-			)
+
+			toTypeName, err := c.getIdentifierName(toFieldType)
 			if err != nil {
 				return err
 			}
+
+			if converterMethodName, ok := c.convStructMap[toFieldType.Name]; ok {
+				// the converter for this struct has already been generated, so just use it
+				c.append(indent, "out.", keyPrefix+pascalCase(toField.Name), " = ", converterMethodName, "(", fromIdentifier, ")")
+			} else if c.useRecurGen && fromTypeName != "" {
+				// generate a callable converter inside function literal
+				err = c.genConverterForStructWrapped(
+					toField,
+					toFieldType,
+					toTypeName,
+					toSubIdentifier,
+					fromTypeName,
+					fromIdentifier,
+					stFromType,
+					fieldMap,
+					prevKeyPrefixes,
+					indent,
+				)
+				if err != nil {
```

Comment:
if you don't handle err, do directly return 

```
return c.genConverterForStructWrapped{
```

and again, should you log / handle here?